### ISSUE TITLE
fix cmsweb port to be 8443 and add proper userAgent reporting in HTTP calls

### ISSUE THREAD

**Issue by belforte** [dmwm/CRABClient#4948]
looks like client connects to REST via URL's like
(7, 'Failed to connect to cmsweb-testbed.cern.ch port 443: Connection refused')

(had to pick a failure message to see the port number !)

But according to current planning we should be using port 8443
https://hypernews.cern.ch/HyperNews/CMS/get/webInterfaces/1750.html

Need to fix this in order to use new Authentication method and cmsweb-auth cluster.

**Comment by belforte** [comment]
main question is why it does not happen transparently via
https://github.com/dmwm/WMCore/blob/b0e11b6987bd57ae156c09232fe4566456a0209c/src/python/WMCore/Services/Service.py#L122-L124
```python
        # setup port 8443 for cmsweb services
        if cfg_dict['endpoint'].startswith("https://cmsweb"):
            cfg_dict['endpoint'] = cfg_dict['endpoint'].replace('.cern.ch/', '.cern.ch:8443/', 1)
```

**Comment by belforte** [comment]
CRABCLient does not use `WMCore/Services/Service.py` not `WMCore/Services/Requests.py` but 
https://github.com/dmwm/CRABServer/blob/master/src/python/RESTInteractions.py
```python
class HTTPRequests(dict):
    """
    This code is a simplified version of WMCore.Services.Requests - we don't
    need all the bells and whistles here since data is always sent via json we
    also move the encoding of data out of the makeRequest.
    HTTPRequests does no logging or exception handling, these are managed by the
    Client class that instantiates it.
    NOTE: This class should be replaced by the WMCore.Services.JSONRequests if WMCore
    is used more in the client.
    """
```

while the wisdom of having a separate version of `WMCore/Services/Requests.py` can surely be questioned and the matter revisited, that is quite some work and better be left for py2/3 migration where all https interactions need to be revised.

For the time being I will port to CRABServer `RestInteractions.py` the port fix from comment above

**Comment by belforte** [comment]
the change in CRABServer will require extensive validation, can't push this is as a critical fix.
See: https://github.com/dmwm/CRABServer/issues/6232#issuecomment-721657241

**Comment by belforte** [comment]
reference on CMSWEB side: https://its.cern.ch/jira/browse/CMSKUBERNETES-44

**Comment by belforte** [comment]
Ready for this. Should be as simple as rebuilding with latest CRABServer once new tag is ready. But should look into using same REST client in getProxiedWebDir or even better remove this call.

**Comment by belforte** [comment]
I was wrong in thinking that CRABInterace/HTTPRequests is only used in tasks.py. The call there is only used in a special circumstance. The main server initialization happens via Emulator.
So currently I should be missing it, but looking at monitoring I see that almost all CRABClient calls now are reported as uding port 8443.
I need to understand what's going on.

**Comment by belforte** [comment]
after some investigation I believe that report of  CRABClient on port 8443 is due to some improper reporting from K8s FE side. See:
 https://its.cern.ch/jira/browse/CMSKUBERNETES-44?focusedCommentId=3667736&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-3667736

**Comment by belforte** [comment]
The problem in monitoring has been confirmed. So I can't check things fully at this moment, but understanding is correct so far.
I will go on with passing a RESTServer initialized by the client to getProxiedWebDir

**Comment by belforte** [comment]
getProxiedWebDir is used in
- [ ] CopyCat.py
- [ ] prepareLocal.py
- [ ] status.py
- [ ] getlog.py
- [ ]

**Comment by belforte** [comment]
Actually, REST Server instantiation in client happens when each command calls `getEmulator` at the beginning like in
https://github.com/dmwm/CRABClient/blob/6e2b72a9ec158412379dc3943bf9c5de48d1855b/src/python/CRABClient/Commands/kill.py#L18-L19
even if a server rest is already instantiated in the common init method of `SubCommand`
https://github.com/dmwm/CRABClient/blob/6e2b72a9ec158412379dc3943bf9c5de48d1855b/src/python/CRABClient/Commands/SubCommand.py#L244
but never used.
So..
- [ ] cleanup and unify usage of restServer in various commands

**Comment by belforte** [comment]
a solution for this is now in my branch https://github.com/belforte/CRABClient/tree/useNewRestInteraction
changes are quite extensive, but now there is almost one single instantiation of CRABRest server
(special case is uploadlog which is currently not a command like the others,  and I have not completed the work on `copyCat` which anyhow we never used and it is a bit tricky).

This requires of course more testing.

Then I need to tackle:
- [ ] crabcache (which is initialized as a UserFileCache instance) but maybe here we can simply change the URL in the REST config
- [ ] access to schedd files via cmsweb*/schedmon i.e. the proxiedWebDir, here it also likely better a solution in the config.

**Comment by belforte** [comment]
the changes for the CRABRest part are in https://github.com/dmwm/CRABClient/pull/4969

**Comment by belforte** [comment]
Now merged and tagged as [v3.210127](https://github.com/dmwm/CRABClient/releases/tag/v3.210127)
